app/models/post.rb

```diff
@@ -1,3 +1,29 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+
+  def initialize(*args)
+    super(*args)
+    update_posts if args.any? # Executes the method only if an argument is provided.
+  end
+
+  # Method to update the posts_counter for a user.
+  def update_posts
+    user = User.find_by(id: author_id)
+    counter = if user.posts_counter.nil?
+                1
+              else
+                user.posts_counter + 1
+              end
+    user.update(posts_counter: counter)
+  end
```

Comment:
- [ ] I suggest that you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database instead of using the initialize method. :relaxed: 
  
  ```ruby
  after_save :update_post_counter
  ```
- [ ] I suggest that you use the `increment` method for the `posts_counter` attribute of a author object. This is useful if you want to keep track of the number of posts created by an author without having to recompute the count every time. 
  
  ```ruby
  def update_post_counter
    author.increment!(:posts_counter)
  end
  ```